toolkit/tools/pkg/imagecustomizerlib/errors.go

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package imagecustomizerlib
+
+import (
+	"errors"
+	"fmt"
+)
+
+// Global error types for categorization
+var (
+	ErrTypeConfigValidation     = errors.New("config-validation")
```

Comment:
I think using a naming pattern like `ConfigValidationError` will be better. Implement this across the types.